rcljava/src/main/cpp/org_ros2_rcljava_node_NodeImpl.cpp

```diff
@@ -410,3 +410,35 @@ Java_org_ros2_rcljava_node_NodeImpl_nativeGetSubscriptionNamesAndTypesByNode(
     rcljava_throw_rclexception(env, ret, "failed to fini subscription names and types structure");
   }
 }
+
+JNIEXPORT void JNICALL Java_org_ros2_rcljava_node_NodeImpl_nativeGetServiceNamesAndTypesByNode(
+  JNIEnv * env, jclass, jlong handle, jstring jname, jstring jnamespace, jobject jnames_and_types)
+{
+  rcl_node_t * node = reinterpret_cast<rcl_node_t *>(handle);
+  if (!node) {
+    rcljava_throw_exception(env, "java/lang/IllegalArgumentException", "node handle is NULL");
+    return;
+  }
+
+  const char * name = env->GetStringUTFChars(jname, NULL);
+  const char * namespace_ = env->GetStringUTFChars(jnamespace, NULL);
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  rcl_names_and_types_t service_names_and_types = rcl_get_zero_initialized_names_and_types();
+
+  rcl_ret_t ret = rcl_get_service_names_and_types_by_node(
+    node,
+    &allocator,
+    name,
+    namespace_,
+    &service_names_and_types);
+  RCLJAVA_COMMON_THROW_FROM_RCL_X(env, ret, "failed to get service names and types", goto cleanup);
```

Comment:
Consider using scoped exit from rcpputils.

Reply:
done in https://github.com/osrf/ros2_java/pull/36/commits/6e804e834fc5817d868c35123a6ce90b01fab3c9.